dxf_api/src/ipf/live/InstrumentProfileConnection.cs

```diff
@@ -352,13 +355,14 @@ private void Download()
                     if (time == LastModified)
                         return; // nothing changed
                     MakeConnected();
-                    using (Stream inputStream = webResponse.GetResponseStream())
+                    using (var inputStream = webResponse.GetResponseStream())
                     {
-                        StreamCompression compress = (isFileStream ? StreamCompression.DetectCompressionByExtension(new Uri(address)) :
-                            StreamCompression.DetectCompressionByMimeType(webResponse.ContentType));
-                        using (Stream decompressedIn = compress.Decompress(inputStream))
-                        {
-                            int count = process(decompressedIn);
+                        var compress = isFileStream ? StreamCompression.DetectCompressionByExtension(new Uri(address)) :
+                            StreamCompression.DetectCompressionByMimeType(webResponse.ContentType);
+                        using (var decompressedIn = compress.Decompress(inputStream)) {
+                            if (!isFileStream) decompressedIn.ReadTimeout = DefaultReadTimeout;
```

Comment:
Правильный путь завершать поток Handler в Close. Таймаут чтения по сути нивелирует саму идею live обновления, заменяя её полингом (при чём в этом случае, возможно, даже без modified since хедера).

Reply:
Судя по всему, код является переписыванием Java и при переписывании забыли слипы и завершение потока, но интересно то, что `Interrupt` не влияет на стримы и сокеты, то есть, исключение, аналогично Java, не бросается. 

Завершение потока получается сымитировать с помощью `Abort`, с пробрасыванием исключения из `InstrumentProfileParser`, но возникает лаг в несколько секунд (какой-то внутренний нерегулируемый таймаут, видимо). Но это не  очень хороший путь. 

Да, видимо придётся пытаться закрывать соединение или искать какой-то другой способ.

Reply:
Использовал закрытие WebResponse